BedrockServer.h

```diff
@@ -8,30 +8,95 @@
+    //    response is sent.
+    // 3. We set timeouts for any HTTPS commands to five seconds. Note that if we are slaving, this has no effect on
+    //    commands that were escalated to master, we continue waiting for those commands.
+    //
+    // The server then continues operating as normal until there are no client connections left (because we've stopped
+    // accepting them, and closed any connections as we responded to their commands), at which point it switches to
+    // CLIENTS_RESPONDED. The sync node notices this, and on it's next update() loop, it begins shutting down. If it
+    // was mastering, the first thing it does in this case is switch to STANDINGDOWN. See more on standing down in the
+    // next section.
+    //
+    // The sync node will continue STANDINGDOWN until two conditions are true:
+    // 
+    // 1. There are no commands in progress.
+    // 2. There are no outstanding queued commands.
+    //
+    // A "command in progress" is essentially any command that we have not completed but is not in our main command
+    // queue. We increment the count of these commands when we dequeue a command from the main queue, and decrememnt it
+    // when we respond to the command (and also in a few other exception cases where the command is abandoned or does
+    // not require a response). This means that if a command has been moved to the queue of outstanding HTTPS commands,
+    // or the sync thread queue, or is currently being handled by a worker, or escalated to master, it's "in progress".
+    //
+    // If we were not mastering, there is no STANDINGDOWN state to wait through - all of a slaves commands come from
+    // local clients, and once those connections are all closed, then that means every command has been responded to,
+    // implying that there are neither queued commands nor commands in progress.
+    //
+    // The sync node then finishes standing down, informing the other nodes in the cluster that it is now in the
+    // searching STATE. At this point, we switch to the DONE state. The sync thread waits for worker threads to join.
+    // The worker threads, when they find the main command queue empty, will check for the DONE state, and return. The
+    // sync thread can then complete and the server is shut down.
+    //
+    //
+    // # Standing Down
+    // Standing down when shutting down is covered in the above section, but there's an additional bit of work to do of
+    // we're standing down without shutting down. The main difference is that we are not waiting on all existing
+    // clients to be disconnected while we stand down - we will be able to service these same clients as a slave as
+    // soon as we finish this operation. However, we still have the same criteria for STANDINGDOWN as listed above, no
+    // commands in progress, and an empty command queue.
+    //
+    // When a node switches to STANDINGDOWN, it starts rejecting escalated commands from peers. This allows it to run
```

Comment:
what happens to these rejected commands?

Reply:
The peers hold on to them, look for a new master, and when they find one, they re-escalate them to the new master.

Reply:
Does this mean peers will back up commands the entire time master is STANDINGDOWN? This means the cluster halts for the entire duration fo STANDINGDOWN, which as described above is until every existing command finishes. What is the longest a command can run? How long is the longest amount of time we'll be stuck in this halted state?

Reply:
This behavior is not new, peers have always looked for a new master and re-escalated commands when master is standing down.

The cluster doesn't halt, escalated commands sit in slave queues instead of the master queue until master finishes standing down. Since master was busty doing things, they would have just been sitting in it's queue waiting to finish, anyway. The exception is for commands that are "stuck" (i.e., with outstanding HTTPS requests that aren't finishing), which is why we shorten all timeouts to 5 seconds while standing down or shutting down.

Note that read commands can still be executed on slaves.

There is no strict limit on how long we can be STANDINGDOWN for. If we stand down when there are 50 million commands in the master queue, it will take a while. However, if we kept escalating commands to master while here were already 50 million commands in that queue, they would still take a while to get processed.